### PR TITLE
Repair the Nix flake

### LINKED ISSUE #4940: Nix flake is invalid
When I ran into the issue #4939 fixes, I started exploring the flake, and noticed some commands failing, notably `nix flake check`.

The flake contains nested outputs, which aren’t allowed.

### PR DESCRIPTION
**Pull request by sellout**
## Overview

This does the minimum to get `nix flake check` working.

The primary issue is that the standard flake outputs require flat package sets, and this flake produced nested ones.

Fixes #4940.

## Implementation notes

This merges package sets into a single one rather than making them separate entries in the outputs.

The only other change was to correct the attribute name used for the UCM docker image’s command.

## Interesting/controversial decisions

This currently flattens the package sets and adds prefixes to them, for some conceptual groupings. The prefixes could be removed, though.

Also, if we wanted to maintain some of the nesting, we could use the `legacyPackages` output, but that is (as it says) legacy, and either way it’s breaking the names, so we might as well just adopt the current flat style.

### CODE REVIEW

**Comment by sellout** [comment]
I can also remove the garnix config before merging. I mostly have it to make sure I’m getting decent coverage on the Nix bits. But it looks like there’s already some of that in GitHub workflows.

**Comment by aryairani** [comment]
> I can also remove the garnix config before merging. I mostly have it to make sure I’m getting decent coverage on the Nix bits. But it looks like there’s already some of that in GitHub workflows.

Just spotted this. Yeah I feel a bit blocked on the garnix piece because I don't understand it and haven't set up a time to go over it.

I'd say let's remove that if it's not integral, and then we can figure out how to improve CI separately, which could include garnix, or sticking with what we have, or replacing some of what we have with something else.

**Comment by sellout** [comment]
> Just spotted this. Yeah I feel a bit blocked on the garnix piece because I don't understand it and haven't set up a time to go over it.
> 
> I'd say let's remove that if it's not integral, and then we can figure out how to improve CI separately, which could include garnix, or sticking with what we have, or replacing some of what we have with something else.

Done.

**Review by aryairani — APPROVED** [review]

**Pull request merged**
